python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/datahub_client.py

```diff
@@ -397,321 +442,108 @@ def upsert_athena_table(
 
-    def search_facets(
+    def _get_custom_property_key_value_pairs(
         self,
-        query: str = "*",
-        result_types: Sequence[ResultType] = (
-            ResultType.DATA_PRODUCT,
-            ResultType.TABLE,
-            ResultType.CHART,
-        ),
-        filters: Sequence[MultiSelectFilter] = (),
-    ) -> SearchFacets:
-        """
-        Returns facets that can be used to filter the search results.
+        custom_properties: CustomEntityProperties,
+    ) -> dict:
         """
-        return self.search_client.search_facets(
-            query=query, result_types=result_types, filters=filters
-        )
-
-    def list_data_product_assets(self, urn, count, start=0) -> SearchResponse:
-        """
-        returns a list of data product children
+        get each custom property as an unnested key/value pair.
+        we cannot push nested structures to datahub custom properties
         """
-        return self.search_client.list_data_product_assets(
-            urn=urn, count=count, start=start
+        custom_properties_dict = json.loads(
+            custom_properties.model_dump_json(), parse_int=str
         )
-
-    def get_glossary_terms(self, count: int = 1000) -> SearchResponse:
-        """Wraps the client's glossary terms query"""
-        return self.search_client.get_glossary_terms(count)
-
-    def get_table_details(self, urn) -> TableMetadata:
-        try:
-            response = self.graph.execute_graphql(self.dataset_query, {"urn": urn})[
-                "dataset"
-            ]
-            properties, custom_properties = parse_properties(response)
-            columns = parse_columns(response)
-            domain = parse_domain(response)
-            owner, owner_email = parse_owner(response)
-            tags = parse_tags(response)
-            name = properties.get("name", response.get("name"))
-            created, modified = parse_created_and_modified(properties)
-
-            # A dataset can't have both a container and data product parent, but if we did
-            # start to use in that we'd need to change this
-            if response["container_relations"]["total"] > 0:
-                relations = parse_relations(
-                    RelationshipType.PARENT, response["container_relations"]
-                )
-            elif response["data_product_relations"]["total"] > 0:
-                relations = parse_relations(
-                    RelationshipType.PARENT, response["data_product_relations"]
-                )
+        custom_properties_unnested = self.unpack(custom_properties_dict)
+        custom_properties_unnested_all_string_values = {
+            key: str(value)
+            # if value is not None else value
+            for key, value in custom_properties_unnested.items()
+        }
+
+        return custom_properties_unnested_all_string_values
+
+    def unpack(self, d, custom_properties=None):
```

Comment:
found this name a bit confusing

```suggestion
        custom_properties_unnested = self._flatten_dict(custom_properties_dict)
        custom_properties_unnested_all_string_values = {
            key: str(value)
            # if value is not None else value
            for key, value in custom_properties_unnested.items()
        }

        return custom_properties_unnested_all_string_values

    def _flatten_dict(self, d, custom_properties=None):
```

Might also be worth moving this to a method of `CustomProperties`